### PR TITLE
Add new `merge_clin()` function for export

### LINKED ISSUE #80: Move `merge_clin.R` functionality from Rscript to an exported function
## Features

- add a new function that does that `merge_clin.R` piece.
- rewrite the merge.clin.R to reference the new functionality
- update the vignette and docs accordingly

Thanks for contributing :partying_face:!

### PR DESCRIPTION
**Pull request by stufield**
# Summary
- new function wrapper to allow users to merge in clinical variables to `soma_adat` objects easily
- closes #80

### CODE REVIEW

File: R/merge-clin.R [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+#' @param clin_data One of 2 options:
+#' \itemize{
+#'   \item a data frame containing clinical variables to merge into `x`, or
+#'   \item a path to a file, typically a `*.csv`,
+#'     containing clinical variables to merge into `x`.
+#' }
+#' @param by A character vector of variables to join by.
+#'   See [dplyr::left_join()] for more details.
+#' @param by_class If `clin_data` is a file path, a named character vector
+#'   of the variable its class. This ensures the "by-key" is compatible
```
Comment (scheidec) [inline R/merge-clin.R:29]: "of the variable _and_ its class"

File: tests/testthat/test-merge-clin.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+# Setup ----
+clin_file <- system.file("cli/merge", "meta.csv", package = "SomaDataIO",
+                          mustWork = TRUE)
+clin_df <- read.csv(clin_file, header = TRUE, colClasses = c(SampleId = "character"))
+apts <- withr::with_seed(123, sample(getAnalytes(example_data), 2L))
+adat <- head(example_data, 9L) |> dplyr::select(SampleId, all_of(apts))
+
+test_that("merge_clin() errors on bad `clin_data` argument", {
```
Comment (scheidec) [inline tests/testthat/test-merge-clin.R:9]: This test description should be modified to something like: `"merge_clin() generates expected merged output"`

File: vignettes/cli-merge-tool.Rmd [hunk 3]
```diff
@@ -39,6 +39,9 @@ in the `cli/merge/` directory, which allows one to
 generate an updated `*.adat` file via the command-line without
 having to launch an integrated development environment ("IDE"), e.g. `RStudio`.
 
+To use `SomaDataIO`s exported functionality from _within_ and R session,
```
Comment (scheidec) [inline vignettes/cli-merge-tool.Rmd:42]: "and" -> "an"

File: vignettes/cli-merge-tool.Rmd [hunk 4]
```diff
@@ -101,12 +106,13 @@ withr::with_dir(analysis_dir,
 ```
 
 
-At this point we have 3 files in a temporary analysis directory:
+At this point we there are now 3 files in our analysis directory:
```
Comment (scheidec) [inline vignettes/cli-merge-tool.Rmd:109]: remove "we"

**Review by scheidec — CHANGES_REQUESTED** [review]
Looks great! Just a few minor grammatical changes suggested.

File: tests/testthat/test-merge-clin.R (hunk 2 above)
Comment (stufield) [reply tests/testthat/test-merge-clin.R:9]: Ohhhhhh, good catch. My copy-paste game was a little too rigorous!

**Review by scheidec — APPROVED** [review]

**Pull request merged**